app/scripts/main.js

```diff
@@ -28,53 +30,51 @@
 
     cameraManager.onframe = function() {
       // There is a frame in the camera, what should we do with it?
- 
-      var imageData = cameraManager.getImageData();
-      var detectedQRCode = qrCodeManager.detectQRCode(imageData, function(url) {
-        if(url !== undefined) {
-          qrCodeManager.showDialog(url);
-        }
-      });
+      if (qrCodeManager.numActiveThreads < qrCodeManager.maxThreads) {
```

Comment:
Nope. The consequence is: The worker won't be called.

On my `i5` it takes about 10ms for the worker to respond back, at the worst case scenario we'll have a `"10ms lag", **` `2.5ms`** on the best case, not noticeable.